models/yield_yak/arbitrum/yield_yak_arbitrum_trades.sql

```diff
@@ -0,0 +1,108 @@
+{{ config(
+    
+    alias = 'trades',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address'],
+    post_hook='{{ expose_spells(\'["arbitrum"]\',
+                                "project",
+                                "yield_yak",
+                                \'["angus_1"]\') }}'
+    )
+}}
+
+
+{% set project_start_date = '2023-03-21' %}
+
+WITH 
+
+dexs as (
+        SELECT 
+            evt_block_time as block_time, 
+            -- '' as taker, commenting this as there's no trader in the event 
+            CAST(NULL as VARBINARY) as maker, 
+            _amountIn as token_sold_amount_raw, 
+            _amountOut as token_bought_amount_raw, 
+            CAST(NULL as double) as amount_usd, 
+            _tokenIn as token_sold_address, 
+            _tokenOut as token_bought_address, 
+            contract_address as project_contract_address, 
+            evt_tx_hash as tx_hash,
+            ARRAY[-1] AS trace_address,
+            evt_index
+        FROM 
+        {{ source('yield_yak_arbitrum', 'YakRouter_evt_YakSwap') }}
+        {% if is_incremental() %}
+        WHERE evt_block_time >= date_trunc('day', now() - interval '7' Day)
+        {% endif %}
```

Comment:
can you update incremental logic to new design pattern?
target like [this](https://github.com/duneanalytics/spellbook/blob/main/models/_sector/dex/trades/avalanche_c/platforms/openocean_avalanche_c_base_trades.sql#L9)
source like [this](https://github.com/duneanalytics/spellbook/blob/main/models/_sector/dex/trades/avalanche_c/platforms/openocean_avalanche_c_base_trades.sql#L30-L32)

Reply:
plz apply across the whole PR

Reply:
Have done this, but was not sure if there was also a way to fit something like the below for the `not is_incremental()` scenario into the new design pattern (?)
```
    {% if not is_incremental() %}
    AND tx.block_time >= TIMESTAMP '{{project_start_date}}'
    {% endif %}
```

Reply:
nope, the `not` portion will remain filtering on hardcoded date variable when project started. it's helpful to filter base tables in joins as much as possible, especially when data in decoded table starts much later than data in base table (which most cases do).

ideal setup:
```
    {% if not is_incremental() %}
    AND tx.block_time >= TIMESTAMP '{{project_start_date}}'
    {% else %}
    AND {{incremental_predicate('tx.block_time')}}
    {% endif %}
```

the if/then can also be flipped, doesn't matter 🤝 
it's cleaner to see the `else` in there vs. having an `endif` and `if` again.

it's not a blocker by any means, so we can proceed as-is, but just a general design pattern for future spells.